sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/exemplar/LongFixedSizeExemplarReservoir.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.exemplar;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.Clock;
+import io.opentelemetry.sdk.metrics.data.LongExemplarData;
+import io.opentelemetry.sdk.metrics.internal.concurrent.AdderUtil;
+import io.opentelemetry.sdk.metrics.internal.concurrent.LongAdder;
+import io.opentelemetry.sdk.metrics.internal.data.ImmutableLongExemplarData;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
+ * Base implementation for fixed-size reservoir sampling of Exemplars.
+ *
+ * <p>Additionally this implementation ONLY exports double valued exemplars.
+ */
+// TODO(anuraaga): Reduce copy-paste from AbstractDoubleFixedSizeExemplarReservoir.
```

Comment:
I copied this for now to keep things straight forward to get the public API in and hope for cleanup in the future in this internal code

Reply:
I have a few ideas on how we could clean this up. I'm fine addressing it in a followup.